packages/astro/src/runtime/server/index.ts

```diff
@@ -208,7 +208,18 @@ Did you mean to add ${formatList(probableRendererNames.map((r) => '`' + r + '`')
 		throw new Error(message);
 	}
 
-	const children = await renderSlot(result, slots?.default);
+	const children: Record<string, string> = {};
+	if (slots) {
+		const promises: Promise<void>[] = []
+		for (const [key, value] of Object.entries(slots)) {
+			promises.push(renderSlot(result, value as string).then((output) => {
+				if (output?.trim() !== '') {
+					children[key] = output;
+				}
+			}))
+		}
+		await Promise.all(promises);
```

Comment:
Rather than just rendering out `children`, we render all `slots`.

Would have been nice to do this lazily/on-demand but most frameworks aren't `async` compatible, unfortunately.